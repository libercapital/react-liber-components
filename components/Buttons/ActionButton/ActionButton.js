'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _StyledButtons = require('../StyledButtons');

var _styles = require('./styles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ActionButton = function (_Component) {
  _inherits(ActionButton, _Component);

  function ActionButton() {
    _classCallCheck(this, ActionButton);

    var _this = _possibleConstructorReturn(this, (ActionButton.__proto__ || Object.getPrototypeOf(ActionButton)).call(this));

    _this.handleOnMouseOver = function () {
      _this.setState({ hover: true });
    };

    _this.handleOnMouseLeave = function () {
      _this.setState({ hover: false });
    };

    _this.buttonSelection = function (type, link) {
      switch (type) {
        case 'success':
          return link ? _StyledButtons.SuccessLink : _StyledButtons.SuccessButton;
        case 'secondary':
          return link ? _StyledButtons.SecondaryLink : _StyledButtons.SecondaryButton;
        case 'primary':
          return link ? _StyledButtons.PrimaryLink : _StyledButtons.PrimaryButton;
        default:
          return link ? _StyledButtons.Link : _StyledButtons.Button;
      }
    };

    _this.state = {
      hover: false
    };
    return _this;
  }

  _createClass(ActionButton, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          type = _props.type,
          icon = _props.icon,
          link = _props.link,
          children = _props.children,
          props = _objectWithoutProperties(_props, ['type', 'icon', 'link', 'children']);

      var hover = this.state.hover;

      var Button = this.buttonSelection(type, link);
      return _react2.default.createElement(
        Button,
        _extends({}, props, {
          onMouseOver: this.handleOnMouseOver,
          onFocus: this.handleOnMouseOver,
          onMouseLeave: this.handleOnMouseLeave,
          onBlur: this.handleOnMouseLeave

        }),
        _react2.default.createElement(
          _styles.Icon,
          { hover: hover },
          icon
        ),
        _react2.default.createElement(
          _styles.Label,
          {
            hover: hover,
            ref: function ref(el) {
              _this2.label = el;
            },
            size: this.label ? this.label.scrollWidth + 'px' : '0px'
          },
          children
        )
      );
    }
  }]);

  return ActionButton;
}(_react.Component);

ActionButton.propTypes = {
  icon: _propTypes2.default.element.isRequired,
  children: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.element, _propTypes2.default.number]),
  type: _propTypes2.default.string,
  link: _propTypes2.default.bool
};

ActionButton.defaultProps = {
  children: null,
  type: null,
  link: false
};

exports.default = ActionButton;