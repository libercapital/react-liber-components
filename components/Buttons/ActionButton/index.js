'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _ActionButton = require('./ActionButton');

var _ActionButton2 = _interopRequireDefault(_ActionButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _ActionButton2.default;