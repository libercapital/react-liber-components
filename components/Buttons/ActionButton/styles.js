'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Icon = exports.Label = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    width: ', ';\n    opacity: ', ';\n  '], ['\n    width: ', ';\n    opacity: ', ';\n  ']),
    _templateObject2 = _taggedTemplateLiteral(['\n    margin-right: ', ';\n  '], ['\n    margin-right: ', ';\n  ']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Util = require('../../Util');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Label = exports.Label = _styledComponents2.default.div.withConfig({
  displayName: 'styles__Label',
  componentId: 'sc-1u3e9hj-0'
})(['text-align:center;width:unset;opacity:1;transition:all 300ms  ', ';white-space:nowrap;color:inherit;', ' ', ''], function (props) {
  return props.hover ? 'ease-out' : 'ease-in';
}, _Util.media.large(_templateObject, function (props) {
  return props.hover ? props.size || '0px' : '0px';
}, function (props) {
  return props.hover ? '1' : '0';
}), _Util.media.extraLarge(_templateObject, function (props) {
  return props.hover ? props.size || '0px' : '0px';
}, function (props) {
  return props.hover ? '1' : '0';
}));

var Icon = exports.Icon = _styledComponents2.default.div.withConfig({
  displayName: 'styles__Icon',
  componentId: 'sc-1u3e9hj-1'
})(['margin-right:15px;transition:margin 300ms  ', ';', ' ', ''], function (props) {
  return props.hover ? 'ease-out' : 'ease-in';
}, _Util.media.large(_templateObject2, function (props) {
  return props.hover ? '15px' : '0px';
}), _Util.media.extraLarge(_templateObject2, function (props) {
  return props.hover ? '15px' : '0px';
}));