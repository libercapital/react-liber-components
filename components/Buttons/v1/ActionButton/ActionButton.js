"use strict";

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.object.assign");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.array.index-of");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.object.keys");

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.object.create");

require("core-js/modules/es6.object.set-prototype-of");

require("core-js/modules/es6.string.link");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _StyledButtons = require("../StyledButtons");

var _styles = require("./styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var ActionButton =
/*#__PURE__*/
function (_Component) {
  _inherits(ActionButton, _Component);

  function ActionButton() {
    var _this;

    _classCallCheck(this, ActionButton);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(ActionButton).call(this));

    _this.handleOnMouseOver = function () {
      _this.setState({
        hover: true
      });
    };

    _this.handleOnMouseLeave = function () {
      _this.setState({
        hover: false
      });
    };

    _this.buttonSelection = function (type, link) {
      switch (type) {
        case 'success':
          return link ? _StyledButtons.SuccessLink : _StyledButtons.SuccessButton;

        case 'secondary':
          return link ? _StyledButtons.SecondaryLink : _StyledButtons.SecondaryButton;

        case 'primary':
          return link ? _StyledButtons.PrimaryLink : _StyledButtons.PrimaryButton;

        default:
          return link ? _StyledButtons.Link : _StyledButtons.Button;
      }
    };

    _this.state = {
      hover: false
    };
    return _this;
  }

  _createClass(ActionButton, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          type = _this$props.type,
          icon = _this$props.icon,
          link = _this$props.link,
          children = _this$props.children,
          props = _objectWithoutProperties(_this$props, ["type", "icon", "link", "children"]);

      var hover = this.state.hover;
      var Button = this.buttonSelection(type, link);
      return _react["default"].createElement(Button, _extends({}, props, {
        onMouseOver: this.handleOnMouseOver,
        onFocus: this.handleOnMouseOver,
        onMouseLeave: this.handleOnMouseLeave,
        onBlur: this.handleOnMouseLeave
      }), _react["default"].createElement(_styles.Icon, {
        hover: hover
      }, icon), _react["default"].createElement(_styles.Label, {
        hover: hover,
        ref: function ref(el) {
          _this2.label = el;
        },
        size: this.label ? "".concat(this.label.scrollWidth, "px") : '0px'
      }, children));
    }
  }]);

  return ActionButton;
}(_react.Component);

ActionButton.propTypes = {
  icon: _propTypes["default"].element.isRequired,
  children: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].element, _propTypes["default"].number]),
  type: _propTypes["default"].string,
  link: _propTypes["default"].bool
};
ActionButton.defaultProps = {
  children: null,
  type: null,
  link: false
};
var _default = ActionButton;
exports["default"] = _default;