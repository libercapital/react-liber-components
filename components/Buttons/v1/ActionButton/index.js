"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ActionButton = _interopRequireDefault(require("./ActionButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _ActionButton["default"];
exports["default"] = _default;