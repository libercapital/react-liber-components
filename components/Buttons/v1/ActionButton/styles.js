"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Icon = exports.Label = void 0;

require("core-js/modules/es6.object.define-properties");

require("core-js/modules/es6.object.freeze");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Util = require("../../../Util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    margin-right: ", ";\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    margin-right: ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    width: ", ";\n    opacity: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    width: ", ";\n    opacity: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Label = _styledComponents["default"].div.withConfig({
  displayName: "styles__Label",
  componentId: "wminmz-0"
})(["text-align:center;width:unset;opacity:1;transition:all 300ms  ", ";white-space:nowrap;color:inherit;", " ", ""], function (props) {
  return props.hover ? 'ease-out' : 'ease-in';
}, _Util.media.large(_templateObject(), function (props) {
  return props.hover ? props.size || '0px' : '0px';
}, function (props) {
  return props.hover ? '1' : '0';
}), _Util.media.extraLarge(_templateObject2(), function (props) {
  return props.hover ? props.size || '0px' : '0px';
}, function (props) {
  return props.hover ? '1' : '0';
}));

exports.Label = Label;

var Icon = _styledComponents["default"].div.withConfig({
  displayName: "styles__Icon",
  componentId: "wminmz-1"
})(["margin-right:15px;transition:margin 300ms  ", ";", " ", ""], function (props) {
  return props.hover ? 'ease-out' : 'ease-in';
}, _Util.media.large(_templateObject3(), function (props) {
  return props.hover ? '15px' : '0px';
}), _Util.media.extraLarge(_templateObject4(), function (props) {
  return props.hover ? '15px' : '0px';
}));

exports.Icon = Icon;