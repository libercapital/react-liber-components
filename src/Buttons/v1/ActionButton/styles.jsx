import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
