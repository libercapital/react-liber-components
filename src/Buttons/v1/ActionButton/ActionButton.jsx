import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button as Btn,
  PrimaryButton as PriBtn,
  SecondaryButton as SecBtn,
  SuccessButton as SucBtn,
  Link as Lnk,
  PrimaryLink as PriLnk,
  SecondaryLink as SecLnk,
  SuccessLink as SucLnk,
} from '../StyledButtons';
import { Label, Icon } from './styles';

class ActionButton extends Component {
  constructor() {
    super();
    this.state = {
      hover: false,
    };
  }

  handleOnMouseOver = () => { this.setState({ hover: true }); }

  handleOnMouseLeave = () => { this.setState({ hover: false }); }

  buttonSelection = (type, link) => {
    switch (type) {
      case 'success':
        return link ? SucLnk : SucBtn;
      case 'secondary':
        return link ? SecLnk : SecBtn;
      case 'primary':
        return link ? PriLnk : PriBtn;
      default:
        return link ? Lnk : Btn;
    }
  }

  render() {
    const {
      type, icon, link, children, ...props
    } = this.props;
    const { hover } = this.state;
    const Button = this.buttonSelection(type, link);
    return (
      <Button
        {...props}
        onMouseOver={this.handleOnMouseOver}
        onFocus={this.handleOnMouseOver}
        onMouseLeave={this.handleOnMouseLeave}
        onBlur={this.handleOnMouseLeave}

      >
        <Icon hover={hover}>{icon}</Icon>
        <Label
          hover={hover}
          ref={(el) => { this.label = el; }}
          size={this.label ? `${this.label.scrollWidth}px` : '0px'}
        >
          {children}
        </Label>
      </Button>
    );
  }
}

ActionButton.propTypes = {
  icon: PropTypes.element.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.number,
  ]),
  type: PropTypes.string,
  link: PropTypes.bool,
};

ActionButton.defaultProps = {
  children: null,
  type: null,
  link: false,
};

export default ActionButton;
