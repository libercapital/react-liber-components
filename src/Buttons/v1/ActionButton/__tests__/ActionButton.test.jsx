import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ActionButton from '../ActionButton';

describe('Test for Action Button', () => {
  configure({ adapter: new Adapter() });
  const props = {
    icon: <div />,
  };
  it('should render correctly', () => {
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when type is primary', () => {
    props.type = 'primary';
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when type is secondary', () => {
    props.type = 'secondary';
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when type is success', () => {
    props.type = 'success';
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when hover', () => {
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    wrapper.simulate('mouseover');
    expect(wrapper.state('hover')).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when focus', () => {
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    wrapper.simulate('focus');
    expect(wrapper.state('hover')).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should change hover when mouseleave', () => {
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    wrapper.setState({ hover: true });
    wrapper.simulate('mouseleave');
    expect(wrapper.state('hover')).toBeFalsy();
  });
  it('should change hover when blur', () => {
    const wrapper = shallow(<ActionButton {...props}>label</ActionButton>);
    wrapper.setState({ hover: true });
    wrapper.simulate('blur');
    expect(wrapper.state('hover')).toBeFalsy();
  });
});
