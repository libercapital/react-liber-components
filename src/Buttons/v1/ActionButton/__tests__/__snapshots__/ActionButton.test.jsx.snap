// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Action Button should render correctly 1`] = `
<StyledButtons__Button
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <styles__Icon
    hover={false}
  >
    <div />
  </styles__Icon>
  <styles__Label
    hover={false}
    size="0px"
  >
    label
  </styles__Label>
</StyledButtons__Button>
`;

exports[`Test for Action Button should render correctly when focus 1`] = `
<StyledButtons__SuccessButton
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <styles__Icon
    hover={true}
  >
    <div />
  </styles__Icon>
  <styles__Label
    hover={true}
    size="0px"
  >
    label
  </styles__Label>
</StyledButtons__SuccessButton>
`;

exports[`Test for Action Button should render correctly when hover 1`] = `
<StyledButtons__SuccessButton
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <styles__Icon
    hover={true}
  >
    <div />
  </styles__Icon>
  <styles__Label
    hover={true}
    size="0px"
  >
    label
  </styles__Label>
</StyledButtons__SuccessButton>
`;

exports[`Test for Action Button should render correctly when type is primary 1`] = `
<StyledButtons__PrimaryButton
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <styles__Icon
    hover={false}
  >
    <div />
  </styles__Icon>
  <styles__Label
    hover={false}
    size="0px"
  >
    label
  </styles__Label>
</StyledButtons__PrimaryButton>
`;

exports[`Test for Action Button should render correctly when type is secondary 1`] = `
<StyledButtons__SecondaryButton
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <styles__Icon
    hover={false}
  >
    <div />
  </styles__Icon>
  <styles__Label
    hover={false}
    size="0px"
  >
    label
  </styles__Label>
</StyledButtons__SecondaryButton>
`;

exports[`Test for Action Button should render correctly when type is success 1`] = `
<StyledButtons__SuccessButton
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <styles__Icon
    hover={false}
  >
    <div />
  </styles__Icon>
  <styles__Label
    hover={false}
    size="0px"
  >
    label
  </styles__Label>
</StyledButtons__SuccessButton>
`;
